pandas/tests/io/formats/style/test_to_latex.py

```diff
@@ -484,8 +484,160 @@ def test_parse_latex_css_conversion(css, expected):
     css = [("command", "option--latex--wrap")]
     expected = [("command", "option--wrap")]
     result = _parse_latex_css_conversion(css)
     assert result == expected
+
+
+def test_longtable_comprehensive(styler):
+    result = styler.to_latex(
+        environment="longtable", hrules=True, label="fig:A", caption=("full", "short")
+    )
+    assert result == dedent(
+        """\
+        \\begin{longtable}{lrrl}
+        \\caption[short]{full} \\label{fig:A} \\\\
+        \\toprule
+        {} & {A} & {B} & {C} \\\\
+        \\midrule
+        \\endfirsthead
+        \\caption[]{full} \\\\
+        \\toprule
+        {} & {A} & {B} & {C} \\\\
+        \\midrule
+        \\endhead
+        \\midrule
+        \\multicolumn{4}{r}{Continued on next page} \\\\
+        \\midrule
+        \\endfoot
+        \\bottomrule
+        \\endlastfoot
+        0 & 0 & -0.61 & ab \\\\
+        1 & 1 & -1.22 & cd \\\\
+        \\end{longtable}
+    """
+    )
+
+
+def test_longtable_minimal(styler):
+    result = styler.to_latex(environment="longtable")
+    assert result == dedent(
+        """\
+        \\begin{longtable}{lrrl}
+        {} & {A} & {B} & {C} \\\\
+        \\endfirsthead
+        {} & {A} & {B} & {C} \\\\
+        \\endhead
+        \\multicolumn{4}{r}{Continued on next page} \\\\
+        \\endfoot
+        \\endlastfoot
+        0 & 0 & -0.61 & ab \\\\
+        1 & 1 & -1.22 & cd \\\\
+        \\end{longtable}
+    """
+    )
+
+
+def test_longtable_multiindex_columns(df):
+    cidx = MultiIndex.from_tuples([("A", "a"), ("A", "b"), ("B", "c")])
+    df.columns = cidx
+    expected = dedent(
+        """\
+        \\begin{longtable}{lrrl}
+        {} & \\multicolumn{2}{r}{A} & {B} \\\\
+        {} & {a} & {b} & {c} \\\\
+        \\endfirsthead
+        {} & \\multicolumn{2}{r}{A} & {B} \\\\
+        {} & {a} & {b} & {c} \\\\
+        \\endhead
+        """
+    )
+    assert expected in df.style.to_latex(environment="longtable")
+
+    # non-sparse
+    expected = dedent(
+        """\
+        \\begin{longtable}{lrrl}
+        {} & {A} & {A} & {B} \\\\
+        {} & {a} & {b} & {c} \\\\
+        \\endfirsthead
+        {} & {A} & {A} & {B} \\\\
+        {} & {a} & {b} & {c} \\\\
+        \\endhead
+        """
+    )
+    assert expected in df.style.to_latex(environment="longtable", sparse_columns=False)
+
+
+def test_longtable_caption_label(styler):
+    expected = dedent(
+        """\
+        \\caption{full} \\\\
+        {} & {A} & {B} & {C} \\\\
+        \\endfirsthead
+        \\caption[]{full} \\\\
+        """
+    )
+    assert expected in styler.to_latex(environment="longtable", caption="full")
+
+    expected = dedent(
+        """\
+        \\caption[short]{full} \\label{fig:A} \\\\
+        {} & {A} & {B} & {C} \\\\
+        \\endfirsthead
+        \\caption[]{full} \\\\
+        """
+    )
+    assert expected in styler.to_latex(
+        environment="longtable", caption=("full", "short"), label="fig:A"
```

Comment:
I guess, it will be more reasonable if the label is called ``tab:A``, to avoid confusion: the environment is a longtable, but the label suggests a figure.

Reply:
done